datafusion/src/execution/context.rs

```diff
@@ -691,6 +693,7 @@ impl Default for ExecutionConfig {
                 Arc::new(SimplifyExpressions::new()),
                 Arc::new(HashBuildProbeOrder::new()),
                 Arc::new(LimitPushDown::new()),
+                // Arc::new(CommonSubexprEliminate::new()),
```

Comment:
I suggest running this CSE immediately after `ConstantFolding` as it may simplify expressions in a way that helps other passes (e.g. make join conditions clearer, or simplify filter expressions so they can be pushed down)

Reply:
Good point, I place it right after the `ConstantFolding` now.